Tests/RxCocoaTests/PublisherTests.swift

```diff
@@ -0,0 +1,147 @@
+//
+//  PublisherTests.swift
+//  Tests
+//
+//  Created by Krunoslav Zaher on 2/26/17.
+//  Copyright © 2017 Krunoslav Zaher. All rights reserved.
+//
+
+import Dispatch
+import RxSwift
+import RxCocoa
+import XCTest
+import RxTest
+
+class PublisherTest: RxTest {}
+
+extension PublisherTest {
+    func testPublisherSharing() {
+        let scheduler = TestScheduler(initialClock: 0)
+        
+        let observer1 = scheduler.createObserver(Int.self)
+        let observer2 = scheduler.createObserver(Int.self)
+        let observer3 = scheduler.createObserver(Int.self)
+        var disposable1: Disposable!
+        var disposable2: Disposable!
+        var disposable3: Disposable!
+        
+        let publisher = Publisher<Int>()
+        scheduler.scheduleAt(100) { publisher.publish(0) }
+        scheduler.scheduleAt(210) { publisher.publish(1) }
+        scheduler.scheduleAt(225) { publisher.publish(2) }
+        scheduler.scheduleAt(245) { publisher.publish(3) }
+        scheduler.scheduleAt(265) { publisher.publish(4)  }
+        
+        scheduler.scheduleAt(200) {
+            disposable1 = publisher.asObservable().subscribe(observer1)
+        }
+        
+        scheduler.scheduleAt(220) {
+            disposable2 = publisher.asObservable().subscribe(observer2)
+        }
+        
+        scheduler.scheduleAt(235) {
+            disposable1.dispose()
+        }
+        
+        scheduler.scheduleAt(260) {
+            disposable2.dispose()
+        }
+        
+        // resubscription
+        
+        scheduler.scheduleAt(260) {
+            disposable3 = publisher.asObservable().subscribe(observer3)
+        }
+        
+        scheduler.scheduleAt(285) {
+            disposable3.dispose()
+        }
+        
+        scheduler.start()
+        
+        XCTAssertEqual(observer1.events, [
+            next(210, 1),
+            next(225, 2)
+            ])
+        
+        XCTAssertEqual(observer2.events, [
+            next(225, 2),
+            next(245, 3)
+            ])
+        
+        XCTAssertEqual(observer3.events, [
+            next(265, 4)
+            ])
+    }
+}
+
+extension PublisherTest {
+    func testEventHubAlwaysObservingOnMainThread() {
```

Comment:
I don't think this unit tests tests anything useful. 

The logic should be:
* subscribe
* send element on background thread to `publisher`
* make sure element is received on main thread and fulfill expectation